src/core.rs

```diff
@@ -41,14 +40,12 @@ impl Core {
         };
 	  }
 
-    pub fn start(&self, serial: Box<dyn SerialPort>, msp_write_delay: Duration, buffer_size: usize) {
-        serial.clear(serialport::ClearBuffer::All).unwrap();
-        let serial_clone = serial.try_clone().unwrap();
+    pub fn start(&self, reader: impl Send + std::io::Read + 'static, writer: impl Send + std::io::Write + 'static, msp_write_delay: Duration, buffer_size: usize) {
```

Comment:
This doesn't look nice at all. Is it necessary? 
I'd rather prefere writing a blanket implementation for your interface on top of something which can do `Read` + `Write`. However, I don't understand the code fully, so take my criticism with a grain of salt.

Reply:
I did it that way because Serialport and TcpStream don't implement the clone trait, so I had to clone them outside the library.

I wish i could see a better solution, but i don't see one at this point

Reply:
Ok, i found the solution in async-msp-cil, so now there is a single `impl Read + Write + Clone`